pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/JavaNode.java

```diff
@@ -4,23 +4,39 @@
 
 package net.sourceforge.pmd.lang.java.ast;
 
+
 import net.sourceforge.pmd.lang.symboltable.Scope;
 import net.sourceforge.pmd.lang.symboltable.ScopedNode;
 
+
+/**
+ * Root interface for all Nodes of the Java AST.
+ */
 public interface JavaNode extends ScopedNode {
 
     /**
-     * Accept the visitor. *
+     * Calls back the visitor's visit method corresponding to the runtime type of this Node.
+     *
+     * @param visitor Visitor to dispatch
+     * @param data    Visit data
      */
     Object jjtAccept(JavaParserVisitor visitor, Object data);
 
+
     /**
-     * Accept the visitor. *
+     * Dispatches the given visitor to the children of this node. This is the default implementation
+     * of {@link JavaParserVisitor#visit(JavaNode, Object)}, to which all other default
+     * implementations for visit methods delegate. Unless visit methods are overridden without calling
+     * {@code super.visit}, the visitor performs a depth-first tree walk.
+     *
+     * <p>The return value of the visit methods called on children are ignored.
+     *
+     * @param visitor Visitor to dispatch
+     * @param data    Visit data
      */
     Object childrenAccept(JavaParserVisitor visitor, Object data);
 
-    @Override
-    Scope getScope();
 
     void setScope(Scope scope);
```

Comment:
I'll mark this also as `@InternalApi` - `setScope()` should be a package private method on one of the abstract base classes of java nodes rather than being a public method on any node...